File: frontend/src/types/schedules.ts
// ============================================================
// DUET CORE - SCHEDULES - TYPES
// ============================================================
//
// Contratos TypeScript utilizados pela área de Agendamentos.
//
// Responsabilidade:
// - representar Robots disponíveis para agendamento;
// - representar Agents disponíveis para agendamento;
// - representar um Schedule retornado pelo Control Room.
//
// Este módulo NÃO:
// - executa chamadas HTTP;
// - contém estado React;
// - possui regras do Scheduler;
// - renderiza componentes.
//
// Os contratos foram extraídos diretamente de
// pages/Schedules.tsx sem alteração dos campos existentes.
// ============================================================


// ============================================================
// ROBOT OPTION
// ============================================================

// Representa um robô disponível para agendamento.
export interface RobotOption {
    id: number;
    name: string;
}


// ============================================================
// AGENT OPTION
// ============================================================

// Representa um Agent disponível para agendamento.
export interface AgentOption {
    agent_id: string;
    name: string;
    status: string;
}


// ============================================================
// SCHEDULE
// ============================================================

// Representa um agendamento retornado pela API.
export interface Schedule {
    id: number;

    robot_id: number;
    robot_name: string;

    agent_id: string | null;
    agent_name: string;

    tipo: string;

    data_inicio: string | null;
    horario: string;

    dias_semana: string | null;

    intervalo_ativo: boolean;
    intervalo_valor: number | null;
    intervalo_unidade: string | null;
    horario_fim: string | null;

    proxima_execucao: string | null;

    ativo: boolean;
}

File: schemas/schedules.py
# ============================================================
# DUET CORE - SCHEMAS - SCHEDULES
# ============================================================
#
# Define os contratos de entrada utilizados pelos endpoints
# HTTP responsáveis pelos agendamentos do Control Room.
#
# Este módulo:
# - contém modelos Pydantic relacionados a Schedules;
# - define os campos aceitos na criação/edição;
# - mantém o contrato atualmente utilizado pela API.
#
# Este módulo NÃO:
# - acessa banco de dados;
# - executa regras do Scheduler;
# - cria threads;
# - registra endpoints FastAPI;
# - executa robôs.
#
# IMPORTANTE:
# Os campos e valores padrão abaixo foram extraídos do
# api/schedules.py sem alteração do contrato atual da API.
# ============================================================

from pydantic import BaseModel


# ============================================================
# MODELO DE CRIAÇÃO / ATUALIZAÇÃO
# ============================================================

class ScheduleCreateRequest(BaseModel):
    """
    Payload utilizado atualmente tanto para criação quanto
    para atualização de um agendamento.
    """

    robot_id: int

    agent_id: str | None = None

    tipo: str

    data_inicio: str

    horario: str

    dias_semana: str | None = None

    intervalo_ativo: bool = False

    intervalo_valor: int | None = None

    intervalo_unidade: str | None = None

    horario_fim: str | None = None